### PR TITLE
Minor improvements, mainly to HIDReport

### PR DESCRIPTION
**Pull request by markuspg**
Hi,

again a minor slew of improvements here and there. I mostly focused on the HIDReport class.

I notice that throughout the project "using namespace std" is used. Is this by design? Generally including namespaces (at least in headers) is considered bad practice, so I'd like to remove these piece by piece. But if it's done on purpose within mfgtools I'd adhere to that in potential future pull requests.

Unfortunately I don't have hardware for testing, so only compilation was checked.

### CODE REVIEW

File: libuuu/cmd.h [hunk 1]
```diff
@@ -45,6 +45,7 @@ using namespace std;
 string get_next_param(const string &cmd, size_t &pos, char sperate = ' ');
 string remove_square_brackets(const string &cmd);
 int get_string_in_square_brackets(const string &cmd, string &context);
+uint16_t str_to_uint16(const std::string &str, bool &conversion_suceeded);
```
Comment (lznuaa) [inline libuuu/cmd.h:48]: I suggest use str_to_uint16(str,  bool *bsuccess=null),  some place don't care resturn value. 
If change str_to_uint16, please also change str_to_uint32

File: libuuu/cmd.cpp [hunk 2]
```diff
@@ -283,6 +283,33 @@ int get_string_in_square_brackets(const string &cmd, string &context)
...
+		const auto tmp_val = std::stoul(str, nullptr, base);
+		if (tmp_val <= UINT16_MAX)
+		{
+			conversion_suceeded = true;
+			return static_cast<uint16_t>(tmp_val);
+		}
+	}  catch (const std::invalid_argument &) {
+	} catch (const std::out_of_range &) {
+	}
+
```
Comment (lznuaa) [inline libuuu/cmd.cpp:309]: Need set_last_err_string before return

File: libuuu/CMakeLists.txt [hunk 3]
```diff
@@ -44,6 +44,7 @@ set(SOURCES
 	tar.cpp
 	rominfo.cpp
 	http.cpp
+	hidreport.cpp
```
Comment (lznuaa) [inline libuuu/CMakeLists.txt:47]: Need update VS project file also. Currently VS file is not from cmake yet.

File: libuuu/hidreport.h [hunk 4]
```diff
@@ -44,24 +44,24 @@ class HIDReport
 
 	size_t get_out_package_size() noexcept { return m_size_out; }
 	virtual void notify(size_t index, uuu_notify::NOTIFY_TYPE type);
-	int read(vector<uint8_t> &buff);
+	int read(std::vector<uint8_t> &buff);
 	void set_notify_total(size_t notify_total) noexcept { m_notify_total = notify_total; }
-	void set_out_package_size(int sz)
+	void set_out_package_size(std::vector<uint8_t>::size_type sz)
 	{
```
Comment (lznuaa) [inline libuuu/hidreport.h:50]: size_t is more simple

**Comment by nxpfrankli** [comment]
Please treat lznuaa as me. I forget switch our github accout

**Pull request merged**